pkg/store/bucket.go

```diff
@@ -442,8 +442,10 @@ func NewBucketStore(
 		return nil, errors.Wrap(err, "validate config")
 	}
 
-	if err := os.MkdirAll(dir, 0750); err != nil {
-		return nil, errors.Wrap(err, "create dir")
+	if dir != "" {
```

Comment:
We could check for `dir == ""` directly here and return if so, I think it's more readable than nested `if`s.

Reply:
Done